src/main/java/edu/wpi/grip/core/SocketHint.java

```diff
@@ -12,44 +13,47 @@
 public class SocketHint<T> {
     public enum View {NONE, SPINNER, SLIDER, RANGE, SELECT}
 
-    private String identifier;
-    private Class<T> type;
-    private View view;
-    private T[] domain;
+    private final String identifier;
+    private final Class<T> type;
+    private final Supplier<T> initialValueSupplier;
```

Comment:
Lets discuss this in person.
Opencv has methods that take null as an allowable param. In some places it would actually be beneficial.